app/models/rook.rb

```diff
@@ -1,3 +1,26 @@
 class Rook < Piece
 
+	def valid_move?(dest_x, dest_y)
+
+		#a move can not be a non-move
+    	return false if dest_x == self.x_position && dest_y == self.y_position
+
+		#no obstructions allowed
+		return false if is_obstructed?(dest_x, dest_y) == true
```

Comment:
Ha! Ok, I can see why you initially thought what you did being that every
piece needs to do that very same thing (DRY). Am I following you?

On Fri, Aug 28, 2015 at 1:56 AM, Stephen B. Thompson <
notifications@github.com> wrote:

> In app/models/rook.rb
> <https://github.com/Rails-Riders/chess/pull/24#discussion_r38172916>:
>
> > @@ -1,3 +1,26 @@
> >  class Rook < Piece
> >
> > +	def valid_move?(dest_x, dest_y)
> > +
> > +		#a move can not be a non-move
> > +    	return false if dest_x == self.x_position && dest_y == self.y_position
> > +
> > +		#no obstructions allowed
> > +		return false if is_obstructed?(dest_x, dest_y) == true
>
> @robgraphix1 <https://github.com/robgraphix1> Whoops! Forget what I just
> said. Sorry, not all my brain's cylinders are firing.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/Rails-Riders/chess/pull/24/files#r38172916>.
>